src/redux/eventSlice.js

```diff
@@ -0,0 +1,114 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+export const getEventAsync = createAsyncThunk(
+  'events/getEventAsync',
+  async () => {
+    const resp = await fetch('http://localhost:3000/v1/events/upcoming');
+    if (resp.ok) {
+      const event = await resp.json();
+      return { event };
```

Comment:
```suggestion
      const json = await resp.json();
      return json;
```
Let's do this for every async functions in here. (return an object with the exact things we need instead of an object containing event.

Reply:
done